web/modules/custom/search_api_field/src/Plugin/Field/FieldWidget/SearchWidget.php

```diff
@@ -0,0 +1,407 @@
+        'id' => 'field-display-overview',
+      ],
+      '#tabledrag' => [
+        [
+          'action' => 'order',
+          'relationship' => 'sibling',
+          'group' => 'field-weight',
+        ],
+        [
+          'action' => 'match',
+          'relationship' => 'parent',
+          'group' => 'field-parent',
+          'subgroup' => 'field-parent',
+          'source' => 'field-name',
+        ],
+      ],
+    ];
+
+    $form['#attached']['library'][] = 'search_api_field/drupal.search_api_field';
+    foreach ($facets as $facet_id => $facet) {
+      $table[$facet_id] = $this->buildFacetRow($facet, $item);
+    }
+    $element['fields'] = $table;
+    $element['refresh_rows'] = ['#type' => 'hidden'];
+    $element['refresh'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Refresh'),
+      '#op' => 'refresh_table',
+      '#submit' => ['::multistepSubmit'],
+      '#ajax' => [
+        'callback' => '::multistepAjax',
+        'wrapper' => 'field-display-overview-wrapper',
+        'effect' => 'fade',
+        // The button stays hidden, so we hide the Ajax spinner too. Ad-hoc
+        // spinners will be added manually by the client-side script.
+        'progress' => 'none',
+      ],
+      '#attributes' => [
+        'class' => [
+          'visually-hidden',
+          'row-refresher',
+        ],
+      ],
+    ];
+
+    $element['query_presets'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Query presets'),
+      '#description' => $this->t('Presets to apply to the query when it is executed. Must be entered in LUCENE syntax.'),
+      '#default_value' => isset($default_values['query_presets']) ? $default_values['query_presets'] : '',
+    ];
+
+    $element['limit'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Limit'),
+      '#description' => $this->t('The number of results to show per page.'),
+      '#default_value' => isset($default_values['limit']) ? $default_values['limit'] : 10,
+      '#min' => 1,
+    ];
+
+    return $element;
+  }
+
+  /**
+   * Returns the region to which a row belongs.
+   *
+   * @param array $row
+   *   The row element.
+   *
+   * @return string|null
+   *   The region name this row belongs to.
+   */
+  public function getRowRegion($row) {
+    return $row['plugin']['type']['#value'];
+  }
+
+  /**
+   * Form submission handler for multistep buttons.
+   */
+  public function multistepSubmit($form, FormStateInterface $form_state) {
+    $trigger = $form_state->getTriggeringElement();
+    $op = $trigger['#op'];
+
+    switch ($op) {
+      case 'edit':
+        // Store the field whose settings are currently being edited.
+        $field_name = $trigger['#field_name'];
+        $form_state->set('plugin_settings_edit', $field_name);
+        break;
+
+      case 'update':
+        // Set the field back to 'non edit' mode, and update $this->entity with
+        // the new settings fro the next rebuild.
```

Comment:
Ahaha, copypaste of web/core/modules/field_ui/src/Form/EntityDisplayFormBase.php:607